webrender/src/device/gfx/device.rs

```diff
@@ -3311,41 +3363,34 @@ impl<B: hal::Backend> Device<B> {
             self.reset_next_frame_resources();
             return;
         }
-        {
-            let image = &self.frames[self.current_frame_id].image;
-            unsafe {
-                if let Some((barrier, pipeline_stages)) = image.transit(
-                    (hal::image::Access::MEMORY_READ, hal::image::Layout::Present),
-                    image.subresource_range.clone(),
-                ) {
-                    self.command_buffer.pipeline_barrier(
-                        pipeline_stages,
-                        hal::memory::Dependencies::empty(),
-                        &[barrier],
-                    );
-                }
-                self.command_buffer.finish();
-            }
-        }
         unsafe {
-            match self.swap_chain.as_mut() {
-                Some(swap_chain) => {
+            let present_pass = self.render_passes.present_pass();
```

Comment:
why are we doing this pass? it seems empty

Reply:
This is just a workaround to fix validation layer errors (that the frame image is not in a presentable layout) until I write the code which identifies the last render pass related to the frame image and we can start this pass there.